### PR TITLE
snap: refresh for core24

### PR DESCRIPTION
**Pull request by Saviq**

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
```
$ snap refresh --channel=24/edge/pr73 ubuntu-frame-osk
error: cannot perform the following tasks:
- Run post-refresh hook of "ubuntu-frame-osk" snap if present (run hook "post-refresh": 
-----
cannot update snap namespace: cannot create symlink in "/usr/share/xml": existing file in the way
snap-update-ns failed with code 1
-----)
```

**Comment by Saviq** [comment]
> ```
> cannot update snap namespace: cannot create symlink in "/usr/share/xml": existing file in the way
> ```

Fixed by `bind`ing instead.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -2,22 +2,23 @@ name: ubuntu-frame-osk
...
+grade: devel
 
 architectures:
   - build-on: amd64
   - build-on: arm64
   - build-on: armhf
 
 plugs:
-  graphics-core22:
+  gpu-snap:
```
Comment (AlanGriffiths) [inline snap/snapcraft.yaml:18]:
```suggestion
  gpu-2024:
```

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -2,22 +2,23 @@ name: ubuntu-frame-osk
...
+grade: devel
 
 architectures:
   - build-on: amd64
   - build-on: arm64
   - build-on: armhf
 
 plugs:
-  graphics-core22:
+  gpu-2024:
```
Comment (AlanGriffiths) [inline snap/snapcraft.yaml:18]:
```suggestion
  gpu-2404:
```

**Comment by AlanGriffiths** [comment]
Now with revision 671:

```
$ WAYLAND_DISPLAY=wayland-2 ubuntu-frame-osk
Setting input sources to [('xkb', 'us')]
xkbcommon: ERROR: failed to add default include path /usr/share/X11/xkb
xkbcommon: ERROR: failed to add default include path /usr/share/X11/xkb
Segmentation fault
```

**Comment by AlanGriffiths** [comment]
OK (modulo the hacks required for the current build chain)

**Review by mattkae — APPROVED** [review]
Working well on my end. I did:

- `snap install ubuntu-frame-osk --channel 24/edge/pr73`
- `ubuntu-frame`
- `ubuntu-frame-osk`
- `gedit`

And I was able to type on the keyboard. I tested both an English and Arabic keyboard

**Review by AlanGriffiths — APPROVED** [review]
WFM

**Pull request merged**